src/libutil/include/nix/util/json-utils.hh

```diff
@@ -24,8 +24,16 @@ nlohmann::json * get(nlohmann::json & map, const std::string & key);
  */
 const nlohmann::json & valueAt(const nlohmann::json::object_t & map, const std::string & key);
 
-std::optional<nlohmann::json> optionalValueAt(const nlohmann::json::object_t & value, const std::string & key);
-std::optional<nlohmann::json> nullableValueAt(const nlohmann::json::object_t & value, const std::string & key);
+/**
+ * @return A pointer to the value assiocated with `key` if `value`
+ * contains `key`, otherwise return  `nullptr` (not JSON `null`!).
+ */
+const nlohmann::json * optionalValueAt(const nlohmann::json::object_t & value, const std::string & key);
+
+/**
+ * Prevents bugs; see `get` for the same trick.
+ */
+const nlohmann::json * optionalValueAt(const nlohmann::json::object_t && value, const std::string & key) = delete;
```

Comment:
Ah, it is the same problem as before. This allowed me to fix it.

(Really, this whole function is just an alias for `get`, and could be deleted.)